### ISSUE TITLE
 Error retrieving client secret from HashiCorp Vault and  [Hashicorp Vault] Failed to resolve secret 'secret1': [Failed to get secret with status 404]

### ISSUE THREAD

**Issue by amnaishaq** [eclipse-tractusx/tractusx-edc#1249]
## Description:

When attempting to run the Docker container edc-controlplane-postgresql-hashicorp-vault:latest with the provided command:

 ```

docker run --name mi-conector -p 8080:8080 -p 8383:8383 -p 8282:8282 -p 9191:9191 -p 9999:9999 -v "$PWD/configuration.properties:/app/configuration.properties" -v "$PWD/logging.properties:/app/logging.properties" -v "$PWD/opentelemetry.properties:/app/opentelemetry.properties" --network my-network edc-controlplane-postgresql-hashicorp-vault:latest
```
```
# Configuración de Vault HashiCorp
edc.vault.hashicorp.url=http://my-vault:8200
edc.vault.hashicorp.token=123456789
edc.vault.hashicorp.secret_path=secret/data/vault/data
edc.vault.hashicorp.secret_key=secret1

# Configuración de la base de datos PostgreSQL
database.url=jdbc:postgresql://localhost:5432/tractusx
database.username=amna
database.password=amna

# Configuración de OAuth2
tx.ssi.oauth.client.id=641c9a45-b3ab-d833-531f-38da819638de
tx.ssi.oauth.client.secret_path=secret/data/vault/data
tx.ssi.oauth.client.secret.alias=secret1
tx.ssi.oauth.client.secret.id=z3iwTVc4G3r1TXhGT4alg9lMsKR8ubWT
tx.ssi.oauth.authorization.url=http://localhost:8200
tx.ssi.oauth.token.url=http://localhost:8200/auth/token


#edc.oauth.token.url=https://dev-qssiii8v6z2dvbe2.us.auth0.com
#edc.oauth.public.key.alias=secret1
#edc.oauth.client.id=z3iwTVc4G3r1TXhGT4alg9lMsKR8ubWT
#edc.oauth.private.key.alias=1vidip12366545fkodojvdknv=
```

```
amna@amna:~/tractusX/tractusx-edc$ vault kv get -address=http://localhost:8200 secret/vault/data
WARNING! VAULT_ADDR and -address unset. Defaulting to https://127.0.0.1:8200.
===== Secret Path =====
secret/data/vault/data

====== Metadata ======
Key              Value
---              -----
created_time     2024-04-22T06:12:36.151065432Z
deletion_time    n/a
destroyed        false
version          4

===== Data =====
Key        Value
---        -----
secret1    DUzJHQTnIHFUsCp5QXvk3BPXtSq1S-xVIF4FO1eFqI9eQjFB9XdCEEebyxfCt3V5
```

 ```

[Hashicorp Vault] Failed to resolve secret 'secret1': [Failed to get secret with status 404]
SEVERE Error booting runtime: java.lang.reflect.InvocationTargetException
´´´

Additionally, the stack trace reveals a NullPointerException with the message "Client secret could not be retrieved".

This issue seems to be related to the retrieval of the client secret from HashiCorp Vault, which is causing the application to fail to boot properly.

##Steps to Reproduce:

1-Run the Docker container edc-controlplane-postgresql-hashicorp-vault:latest with the provided command.
2-Observe the error mentioned above.
##Expected Behavior:

The Docker container should boot successfully without encountering any errors related to the retrieval of secrets from HashiCorp Vault.



````
  ##LOGS
amna@amna:~/tractusX/tractusx-edc$ docker run --name mi-conector -p 8080:8080 -p 8383:8383 -p 8282:8282 -p 9191:9191 -p 9999:9999 -v "$PWD/configuration.properties:/app/configuration.properties" -v "$PWD/logging.properties:/app/logging.properties" -v "$PWD/opentelemetry.properties:/app/opentelemetry.properties" --network my-network edc-controlplane-postgresql-hashicorp-vault:latest
OpenJDK 64-Bit Server VM warning: Sharing is only supported for boot loader classes because bootstrap classpath has been appended
[otel.javaagent 2024-04-23 09:35:13:316 +0000] [main] INFO io.opentelemetry.javaagent.tooling.VersionLogger - opentelemetry-javaagent - version: 1.32.0
INFO 2024-04-23T09:35:16.829484311 Initialized FS Configuration
WARNING 2024-04-23T09:35:16.846279437 The runtime is configured as an anonymous participant. DO NOT DO THIS IN PRODUCTION.
SLF4J(W): No SLF4J providers were found.
SLF4J(W): Defaulting to no-operation (NOP) logger implementation
SLF4J(W): See https://www.slf4j.org/codes.html#noProviders for further details.
INFO 2024-04-23T09:35:17.485248102 Initialized Micrometer Metrics
INFO 2024-04-23T09:35:17.486567477 Initialized Boot Services
INFO 2024-04-23T09:35:17.877318056 Initialized Core Default Services
INFO 2024-04-23T09:35:18.062799777 HTTPS enforcement it not enabled, please enable it in a production environment
INFO 2024-04-23T09:35:18.085194682 HTTPS enforcement it not enabled, please enable it in a production environment
INFO 2024-04-23T09:35:18.106098351 Initialized Hashicorp Vault
INFO 2024-04-23T09:35:18.159249661 Initialized Core Services
WARNING 2024-04-23T09:35:18.177948446 Settings: No setting found for key 'edc.hostname'. Using default value 'localhost'
INFO 2024-04-23T09:35:18.195692694 Initialized JSON-LD Extension
WARNING 2024-04-23T09:35:18.458159829 Error caching context URL 'https://w3id.org/dspace/2024/1/context.json' for URI 'jar:file:/app/edc-controlplane.jar!/document/dspace.jsonld'. Subsequent attempts to expand this context URL may fail.
INFO 2024-04-23T09:35:18.465233981 Initialized Local Transaction
INFO 2024-04-23T09:35:18.465773341 Initialized SQL Core
INFO 2024-04-23T09:35:18.476001026 Initialized org.eclipse.edc.connector.store.sql.assetindex.SqlAssetIndexServiceExtension
INFO 2024-04-23T09:35:18.505123695 Initialized org.eclipse.edc.connector.store.sql.contractnegotiation.SqlContractNegotiationStoreExtension
INFO 2024-04-23T09:35:18.512367757 Initialized org.eclipse.edc.connector.store.sql.policydefinition.SqlPolicyStoreExtension
INFO 2024-04-23T09:35:18.514309361 Initialized Control Plane Default Services
INFO 2024-04-23T09:35:18.519992481 Initialized org.eclipse.edc.connector.store.sql.contractdefinition.SqlContractDefinitionStoreExtension
INFO 2024-04-23T09:35:18.521208912 Initialized org.eclipse.edc.connector.contract.ContractNegotiationDefaultServicesExtension
INFO 2024-04-23T09:35:18.738042968 Initialized Jetty Service
INFO 2024-04-23T09:35:18.747942656 Initialized Jersey Web Service
WARNING 2024-04-23T09:35:18.750405941 Settings for [web.http.protocol.path] and/or [web.http.protocol.path] were not provided. Using default value(s) instead.
INFO 2024-04-23T09:35:18.752022025 Protocol API will be available under port=8282, path=/api/v1/dsp
INFO 2024-04-23T09:35:18.78735752 Initialized Dataspace Protocol API Configuration Extension
INFO 2024-04-23T09:35:18.811707861 Initialized Contract Core
INFO 2024-04-23T09:35:18.81545656 Initialized Contract Negotiation command handlers
INFO 2024-04-23T09:35:18.827075773 Initialized org.eclipse.edc.connector.store.sql.transferprocess.SqlTransferProcessStoreExtension
INFO 2024-04-23T09:35:18.827996609 Initialized Transfer Process Default Services
INFO 2024-04-23T09:35:18.877446702 Initialized Transfer Core
INFO 2024-04-23T09:35:18.881171693 Initialized org.eclipse.edc.connector.transfer.TransferProcessCommandExtension
WARNING 2024-04-23T09:35:18.882037245 No 'name' setting in group edc.datasource.asset found, no schema migrations will run for subsystem asset
INFO 2024-04-23T09:35:18.883647377 Initialized Postgresql schema migration for subsystem asset
WARNING 2024-04-23T09:35:18.885826176 No 'name' setting in group edc.datasource.contractdefinition found, no schema migrations will run for subsystem contractdefinition
INFO 2024-04-23T09:35:18.886995549 Initialized Postgresql schema migration for subsystem contractdefinition
WARNING 2024-04-23T09:35:18.887609501 No 'name' setting in group edc.datasource.contractnegotiation found, no schema migrations will run for subsystem contractnegotiation
INFO 2024-04-23T09:35:18.888150829 Initialized Postgresql schema migration for subsystem contractnegotiation
WARNING 2024-04-23T09:35:18.889846975 No 'name' setting in group edc.datasource.policy found, no schema migrations will run for subsystem policy
INFO 2024-04-23T09:35:18.890160109 Initialized Postgresql schema migration for subsystem policy
WARNING 2024-04-23T09:35:18.890767516 No 'name' setting in group edc.datasource.policy-monitor found, no schema migrations will run for subsystem policy-monitor
INFO 2024-04-23T09:35:18.891057132 Initialized Postgresql schema migration for subsystem policy-monitor
WARNING 2024-04-23T09:35:18.891516466 No 'name' setting in group edc.datasource.transferprocess found, no schema migrations will run for subsystem transferprocess
INFO 2024-04-23T09:35:18.891806838 Initialized Postgresql schema migration for subsystem transferprocess
WARNING 2024-04-23T09:35:18.894880103 No 'name' setting in group edc.datasource.edr found, no schema migrations will run for subsystem edr
INFO 2024-04-23T09:35:18.895574482 Initialized Postgresql schema migration for subsystem edr
WARNING 2024-04-23T09:35:18.896080361 No 'name' setting in group edc.datasource.bpn found, no schema migrations will run for subsystem bpn
INFO 2024-04-23T09:35:18.896328744 Initialized Postgresql schema migration for subsystem bpn
INFO 2024-04-23T09:35:18.899934106 Initialized SQL Business Partner Store
INFO 2024-04-23T09:35:18.904417784 Initialized EDR Cache Core
INFO 2024-04-23T09:35:18.90508401 Initialized OAuth2 Client
INFO 2024-04-23T09:35:18.909324706 Initialized SSI MIW OAuth2 Client
DEBUG 2024-04-23T09:35:19.034557661 [Hashicorp Vault] Failed to resolve secret 'secret1': [Failed to get secret with status 404]
SEVERE 2024-04-23T09:35:19.037641121 Error booting runtime: java.lang.reflect.InvocationTargetException
org.eclipse.edc.spi.system.injection.EdcInjectionException: java.lang.reflect.InvocationTargetException
        at org.eclipse.edc.spi.system.injection.ProviderMethod.invoke(ProviderMethod.java:66)
        at org.eclipse.edc.boot.system.injection.lifecycle.RegistrationPhase.invokeAndRegister(RegistrationPhase.java:51)
        at org.eclipse.edc.boot.system.injection.lifecycle.RegistrationPhase.lambda$invokeProviderMethods$0(RegistrationPhase.java:45)
        at java.base/java.lang.Iterable.forEach(Unknown Source)
        at org.eclipse.edc.boot.system.injection.lifecycle.RegistrationPhase.invokeProviderMethods(RegistrationPhase.java:45)
        at org.eclipse.edc.boot.system.injection.lifecycle.ExtensionLifecycleManager.provide(ExtensionLifecycleManager.java:70)
        at java.base/java.util.stream.ReferencePipeline$3$1.accept(Unknown Source)
        at java.base/java.util.stream.ReferencePipeline$3$1.accept(Unknown Source)
        at java.base/java.util.stream.ReferencePipeline$3$1.accept(Unknown Source)
        at java.base/java.util.stream.ReferencePipeline$3$1.accept(Unknown Source)
        at java.base/java.util.ArrayList$ArrayListSpliterator.forEachRemaining(Unknown Source)
        at java.base/java.util.stream.AbstractPipeline.copyInto(Unknown Source)
        at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(Unknown Source)
        at java.base/java.util.stream.ReduceOps$ReduceOp.evaluateSequential(Unknown Source)
        at java.base/java.util.stream.AbstractPipeline.evaluate(Unknown Source)
        at java.base/java.util.stream.ReferencePipeline.collect(Unknown Source)
        at org.eclipse.edc.boot.system.ExtensionLoader.bootServiceExtensions(ExtensionLoader.java:67)
        at org.eclipse.edc.boot.system.runtime.BaseRuntime.bootExtensions(BaseRuntime.java:139)
        at org.eclipse.edc.boot.system.runtime.BaseRuntime.boot(BaseRuntime.java:200)
        at org.eclipse.edc.boot.system.runtime.BaseRuntime.boot(BaseRuntime.java:83)
        at org.eclipse.edc.boot.system.runtime.BaseRuntime.main(BaseRuntime.java:71)
Caused by: java.lang.reflect.InvocationTargetException
        at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(Unknown Source)
        at java.base/java.lang.reflect.Method.invoke(Unknown Source)
        at org.eclipse.edc.spi.system.injection.ProviderMethod.invoke(ProviderMethod.java:61)
        ... 20 more
Caused by: java.lang.NullPointerException: Client secret could not be retrieved
        at java.base/java.util.Objects.requireNonNull(Unknown Source)
        at org.eclipse.tractusx.edc.iam.ssi.miw.SsiMiwOauth2ClientExtension.createConfiguration(SsiMiwOauth2ClientExtension.java:75)
        at org.eclipse.tractusx.edc.iam.ssi.miw.SsiMiwOauth2ClientExtension.oauth2Client(SsiMiwOauth2ClientExtension.java:67)
        ... 23 more
[otel.javaagent 2024-04-23 09:35:19:149 +0000] [OkHttp http://localhost:4317/...] WARN io.opentelemetry.exporter.internal.grpc.GrpcExporter - Failed to export spans. Server responded with gRPC status code 2. Error message: Failed to connect to localhost/[0:0:0:0:0:0:0:1]:4317
```

**Comment by paullatzelsperger** [comment]
On first sight, I see a couple of issues:
1. secrets path inside vault: in the supposed config you have `secret/data/vault/data` but when you create the secrets you use `secret/vault/data`
2. networking: in your docker command you use a network `my-network`, but the vault seems to be running on your host machine, since you access it with `vault kv`. By default, docker can't access the host's network, unless you provide the `--network host` parameter.
3. This is not an officially supported deployment scenario, so our abilities to support here (e.g. by debugging your local network setup) are quite limited.

TL;DR: the problem is that EDC can't find the secret in the vault, either because it can't access the vault, or because the secret has been stored in the wrong place, or both. 


Also, - and this is the [second time](https://github.com/eclipse-edc/Connector/issues/4127#issuecomment-2068594009) I'm telling you this - please format your issue description properly. You're randomly mixing text with code fragments, which makes it very hard to read. It costs you virtually nothing to tap the "Preview" button and double-check.

**Comment by amnaishaq** [comment]
@paullatzelsperger  The secret is well stored in the vault but I have already tried many ways but it won't let me access it, I always get the same 404 error.
![2024-04-23 13_25_40-Window](https://github.com/eclipse-tractusx/tractusx-edc/assets/127303969/33f07e85-ee74-4aed-879f-fe5206bed2b6)

**Comment by paullatzelsperger** [comment]
> `edc.vault.hashicorp.secret_path=secret/data/vault/data`

that is a different path. also, the property is called `edc.vault.hashicorp.api.secret.path`, not `edc.vault.hashicorp.api.secret_path`

**Comment by paullatzelsperger** [comment]
I recommend you check out the official [helm charts](https://github.com/eclipse-tractusx/tractusx-edc/blob/main/charts/tractusx-connector/README.md) of Tractus-X EDC and go from there. As an alternative, you could check out the [MXD project](https://github.com/eclipse-tractusx/tutorial-resources). Also checkout the [KITs](https://eclipse-tractusx.github.io/docs-kits/kits/Connector%20Kit/Adoption%20View/connector_kit_adoption_view).

I'm sure you understand we simply don't have the bandwidth to help individual users with their non-standard dev setups.

**Comment by paullatzelsperger** [comment]
maybe this is related: https://github.com/eclipse-edc/Connector/discussions/4135
